File: backend/flexikmservlet/src/main/java/alexpre/flexikm/servlet/service/GaranziaService.java
package alexpre.flexikm.servlet.service;

import java.util.List;

import org.springframework.stereotype.Service;

import alexpre.flexikm.servlet.model.Garanzia;

@Service
public interface GaranziaService
{

  public Garanzia insert(Garanzia garanzia);

  public List<Garanzia> getGaranzie();

  public Garanzia update(Garanzia garanzia);

  public String deleteGaranziaById(Long id);

  public List<Garanzia> getGaranziaById(Long id);

}

File: frontend/src/app/services/garanzia.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Garanzia } from '../models/garanzia.model';

@Injectable({ providedIn: 'root' })
export class GaranziaService {
  private baseUrl = '/Garanzia';
  constructor(private http: HttpClient) {}

  getAll(): Observable<Garanzia[]> {
    return this.http.get<Garanzia[]>(`${this.baseUrl}/getGaranzie`);
  }


  getById(id: number): Observable<Garanzia> {
    return this.http.get<Garanzia>(`${this.baseUrl}/getGaranziaById/${id}`);
  }


  create(g: Garanzia): Observable<Garanzia> {
    return this.http.post<Garanzia>(`${this.baseUrl}/insGaranzia`, g);
  }


  update(g: Garanzia): Observable<Garanzia> {
    return this.http.put<Garanzia>(`${this.baseUrl}/updGaranzia`, g);
  }
  

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/deleteGaranziaById/${id}`);
  }
}